src/test/rules/no-classfield-shadowing_test.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Disallows properties shadowed as class field
+ * @author Michel Langeveld <https://github.com/michellangeveld>
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+import rule = require('../../rules/no-classfield-shadowing');
+import {RuleTester} from 'eslint';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    sourceType: 'module',
+    ecmaVersion: 'latest'
+  }
+});
+
+ruleTester.run('no-classfield-shadowing', rule, {
+  valid: [
+    `class MyElement extends LitElement {
+      static properties = {
+        foo: { type: String }
+      }
+    }`,
+    `class MyElement extends LitElement {
+      foo;
+      properties = {
+        foo: { type: String }
+      }
+    }`
+  ],
+
+  invalid: [
+    {
+      code: `class MyElement extends LitElement {
+        foo;
+        static properties = {foo: {}}
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'}
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        static properties = {foo: {}}
+        foo;
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'}
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        foo;
+        static get properties() { return { foo: {}}};
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'}
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        static get properties() { return { foo: {}}};
+        foo;
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'}
+        }
+      ]
+    }
```

Comment:
we should also add a test case with decorators

you can see an example here:
https://github.com/43081j/eslint-plugin-lit/blob/0693199536b6e60cb3088a3909efb943ba0ba41e/src/test/rules/no-property-change-update_test.ts#L60-L71C7

Reply:
Did add this testcaase